sdk/core/azure-core/inc/azure/core/exception.hpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Define RequestFailedException. It is used by HTTP exceptions.
+ */
+
+#pragma once
+
+#include <stdexcept>
+
+namespace Azure { namespace Core {
+  /**
+   * @brief An error while trying to send a request to Azure service.
+   *
+   */
+  struct RequestFailedException : public std::runtime_error
```

Comment:
Suggestion: declare all exceptions as `class` and not `struct` (although it is C++, so it is all legal):
1. It is strange to have struct with virtual destructor (introduced in `std::exception`) and rich inheritance hierarchy.
2. `std::runtime_error` is class, it is strange to have struct deriving from class.
3. Imagine in the future we want to add private fields - it is also strange to have private members in struct.
4. Let's say, you change it to class to the future, but customer has it only forward-declared, and not included i.e. `struct RequestFailedException;` Then, when compiler sees it being forward-declared as struct,but actually being a class, some compilers will generate L4 warning.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/828